drivers/uktty/ns16550/ns16550.c

```diff
@@ -126,35 +128,100 @@ static void init_ns16550(__u64 base)
 	ns16550_reg_write(NS16550_FCR_OFFSET,
 			  ns16550_reg_read(NS16550_FCR_OFFSET) &
 					 ~(NS16550_FCR_FIFO_EN));
+	return 0;
 }
 
-void ns16550_console_init(const void *dtb)
+int ns16550_early_init(void)
+{
+#if CONFIG_LIBUKTTY_NS16550_EARLY_CONSOLE
+	struct ukplat_memregion_desc mrd = {0};
+	struct ukplat_bootinfo *bi;
+	int rc;
+
+	ns16550_uart_base = CONFIG_LIBUKTTY_NS16550_EARLY_CONSOLE_BASE;
+
+	/* From this point we can print */
+	ns16550_uart_initialized = 1;
+
+	bi = ukplat_bootinfo_get();
+	if (unlikely(!bi)) {
```

Comment:
Hmm, should probably just have an assertion here, as the getter already checks stuff.

Reply:
Generally speaking I'd rather not make assumptions on a function's implementation and rather rely on the interface, given that if the implementation changes all calling code breaks. Besides, from a quick look I can see several places where we check the result of `ukplat_bootinfo_get()` rather than asserting.

Reply:
I feel like the problem here may also lie in the fact that it's kinda impossible for this getter to fail (or is it?). Maybe a dtb getter should exist and be used instead. Anyway no matter, too much of a trivial thing that will get overwritten by rearch/refactor anyway. I will let you choose whatever feels more sane here.